File: Intersection_of_Two_Linked_Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        if(!headA || !headB) return NULL;
        
        unordered_set<ListNode*> nodeset;
        while(headA){
            nodeset.insert(headA);
            headA = headA -> next;
        }
        while(headB){
            unordered_set<ListNode*>:: const_iterator it = nodeset.find(headB);
            if (it != nodeset.end())
                return *it;
            headB = headB -> next;
        }
        
        return NULL;
    }
};

File: Intersection_of_Two_Linked_Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        
        listset = set()
        cur = headA
        
        
        while cur:
            listset.add(cur)
            cur = cur.next
        cur = headB
        while cur:
            if cur in listset:
                return cur
            cur = cur.next
        
        
                
        return None